sre-recipes/recipe_runner.py

```diff
@@ -0,0 +1,264 @@
+from recipes.impl_based.base import BaseRecipeImpl
+
+
+# Default Load Generation Config
+DEFAULT_LOADGEN_USER_TYPE = "BasicHomePageViewingUser"
+DEFAULT_LOADGEN_USER_COUNT = 20
+DEFAULT_LOADGEN_SPAWN_RATE = 1
+DEFAULT_LOADGEN_TIMEOUT_SECONDS = 600
+
+
+class ImplBasedRecipeRunner:
+    """A SRE Recipe runner for running recipes implemented as class objects.
+
+    Given a `recipe_name`, it tries to run `recipes/impl_based/recipe_name.py`.
+
+    This runner will propgate all exceptions to the caller, and it is caller's
+    responsibility to handle any exception and to perform any error logging.
+    """
+
+    def __init__(self, recipe_name):
+        self.recipe = None
+        module = importlib.import_module(f"recipes.impl_based.{recipe_name}")
+        for attribute_name in dir(module):
+            attr = getattr(module, attribute_name)
+            if isclass(attr) and attr is not BaseRecipeImpl and issubclass(attr, BaseRecipeImpl):
+                self.recipe = attr()
+                break
+        if not self.recipe:
+            raise NotImplementedError(
+                f"No valid implementation exists for `{recipe_name}` recipe.")
+
+    def get_name(self):
+        return self.recipe.get_name()
+
+    def get_description(self):
+        return self.recipe.get_description()
+
+    def run_break(self):
+        return self.recipe.run_break()
+
+    def run_restore(self):
+        return self.recipe.run_restore()
+
+    def run_hint(self):
+        return self.recipe.run_hint()
+
+    def run_verify(self):
+        return self.recipe.run_verify()
+
+
+class ConfigBasedRecipeRunner:
+    """A SRE Recipe runner for running recipes implemented using configs.
+
+    Given a `recipe_name`, it tries to load `recipes/configs_based/recipe_name.yaml`.
+
+    This runner will propgate all exceptions to the caller, and it is caller's
+    responsibility to handle any exception and to perform any error logging.
+    """
+
+    def __init__(self, recipe_name):
+        filepath = path.join(path.dirname(
+            path.abspath(__file__)), f"recipes/configs_based/{recipe_name}.yaml")
+        with open(filepath, "r") as file:
+            self.recipe = yaml.safe_load(file.read())
+        if not self.recipe:
+            raise ValueError("Cannot parse config as YAML.")
+        self.action_handler = ActionHandler()
+
+    def get_name(self):
+        return self.recipe.get("name", "No name found")
+
+    def get_description(self):
+        return self.recipe.get("description", "No description found")
+
+    @property
+    def config(self):
+        return self.recipe.get("config", {})
+
+    def run_break(self):
+        print('Deploying broken service...')
+        for action in self.config.get("break", []):
+            self.action_handler.handle_action(action)
+        print('Done. Deployed broken service')
+
+    def run_restore(self):
+        print('Restoring broken service...')
+        for action in self.config.get("restore", []):
```

Comment:
can we extract all the operation names into consts so it's consolidated in one place?